metadata-ingestion/src/datahub/ingestion/source/tableau.py

```diff
@@ -1417,6 +1526,7 @@ def emit_custom_sql_datasources(self) -> Iterable[MetadataWorkUnit]:
 
             datasource_name = None
             project = None
+            columns = []
```

Comment:
This is an issue I fixed along the way. We had ingestion runs that failed because the variable `columns` was not defined but used further down. Before, it was only defined if `len(csql[c.DATA_SOURCES]) > 0` and for some reason that wasn't always the case for us. Let me know what you think.